entrypoint_alpine.sh

```diff
@@ -45,6 +45,45 @@ case ${OPENHAB_VERSION} in
         cp -av "${APPDIR}/userdata.dist/." "${APPDIR}/userdata/"
       fi
 
+      # Upgrade userdata if versions do not match
+      curVersion=$(< ${APPDIR}/userdata/etc/version.properties grep build-no | cut -d : -f 2 | tr -d '[:space]')
+      imgVersion=$(< ${APPDIR}/userdata.dist/etc/version.properties grep build-no | cut -d : -f 2 | tr -d '[:space]')
+      
+      if [ "${curVersion}" != "${imgVersion}" ]; then
+        echo "Image build number \"${imgVersion}\" is different from userdata build number \"${curVersion}\""
+
+        # Make a backup of userdata
+        backupFile=userdata-$(date +"%FT%H:%M:%S").tar
+        tar -c -f "/tmp/${backupFile}" -X "${APPDIR}/userdata/backup" "${APPDIR}/userdata"
```

Comment:
When the backup dir does not exist the container fails to start on Alpine with:

```
openhab-upgrade | + tar -c -f /tmp/userdata-2018-01-04T21:00:58.tar -X /openhab/userdata/backup /openhab/userdata
openhab-upgrade | tar: /openhab/userdata/backup: No such file or directory
```
So making sure the dir exists will fix it, e.g. add: `mkdir -p "${APPDIR}/userdata/backup"`